pkg/subscriptions/subscriptions.js

```diff
@@ -0,0 +1,276 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* global jQuery   */
+/* global cockpit  */
+/* global po */
+
+define([
+    "jquery",
+    "latest/cockpit",
+    "latest/po",
+    "latest/mustache"
+], function($, cockpit, po, Mustache) {
+    "use strict";
+
+    cockpit.locale(po);
+
+    function debug() {
+        if (window.debugging == "all" || window.debugging == "subscriptions")
+            console.debug.apply(console, arguments);
+    }
+
+    var subscriptions = { };
+
+    PageSubscriptionsRegister.prototype = {
```

Comment:
While it's not a big deal, you don't actually need to create class here at all. Just run code in the various event handlers like ```dlg.on('shown.bs.modal', function() { /* here */});````